kinto/views/admin.py

```diff
@@ -0,0 +1,134 @@
+
+        # Deleting a bucket cascades to deleting collections and
+        # groups, etc. Fetch all the to-be-deleted objects so we can
+        # pass them correctly to plugins.
+        # FIXME: some more flexible mechanism for getting this info?
+        for (parent_resource, child_resource) in [
+                ('bucket', 'collection'),
+                ('bucket', 'group'),
+                ('collection', 'record'),
+        ]:
+            if parent_resource not in objects_by_resource:
+                continue
+            objects = objects_by_resource[parent_resource]
+            for (object_uri, matchdict) in objects:
+                parent_matchdict = {**matchdict}
+                parent_id = parent_matchdict.pop('id')
+                parent_matchdict[parent_resource + '_id'] = parent_id
+                for child in paginated(storage, sorting=[Sort('id', 1)],
+                                       parent_id=object_uri,
+                                       collection_id=child_resource):
+                    child_matchdict = {'id': child['id'], **parent_matchdict}
+                    child_uri = core_utils.instance_uri(self.request, child_resource,
+                                                        **child_matchdict)
+                    objects_by_resource.setdefault(child_resource, []).append(
+                        (child_uri, child_matchdict))
+
+        print("Expanded into", objects_by_resource)
+        for resource_name, objects in objects_by_resource.items():
+            # Group by parent_uri to make fewer requests to storage backend.
+            objects_by_parent_uri = {}
+            for object, matchdict in objects:
+                path = object.rsplit('/', 2)
+                if len(path) < 2:
+                    parent_uri = ''
+                else:
+                    parent_uri = path[0]
+                objects_by_parent_uri.setdefault(parent_uri, set()).add(matchdict['id'])
+            # FIXME: Deletes are paginated too.. ugh
+            for parent_id, object_ids in objects_by_parent_uri.items():
+                batch_size = 100
+                for batch in slice_into_batches(object_ids, batch_size):
+                    batch = list(batch)
+                    filters = [Filter('id', batch, core_utils.COMPARISON.IN)]
+                    records, _ = storage.get_all(
+                        collection_id=resource_name, parent_id=parent_id,
+                        filters=filters
+                    )
+                    tombstones = storage.delete_all(
+                        collection_id=resource_name, parent_id=parent_id,
+                        filters=filters
+                    )
+                    # FIXME: where do I get this from?
+                    timestamp = None
+                    notify_resource_event(self.request, parent_id, timestamp,
+                                          tombstones, ACTIONS.DELETE, old=records)
```

Comment:
There's definitely some bugs here. In particular, the `quota` plugin fails hilariously to deal with what I'm doing here, and I'm not even sure the `buckets` / `collections` listeners are being fired (because they should have deleted the quota records).